WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/builders/BackupDownloadStateListItemBuilder.kt

```diff
@@ -229,4 +232,44 @@ class BackupDownloadStateListItemBuilder @Inject constructor(
             label = UiStringRes(labelRes),
             itemBottomMarginResId = itemBottomMarginResId
     )
+
+    fun updateCheckboxes(
+        uiState: BackupDownloadUiState,
+        itemType: JetpackAvailableItemType
+    ): List<JetpackListItemState> {
+        var checkedCount = 0
+        val updateCheckboxes = uiState.items.map { state ->
+                if (state.type == CHECKBOX) {
+                    state as CheckboxState
+                    if (state.availableItemType == itemType) {
+                        if (!state.checked) {
+                            checkedCount++
+                        }
+                        state.copy(checked = !state.checked)
+                    } else {
+                        if (state.checked) {
+                            checkedCount++
+                        }
+                        state
+                    }
+                } else {
+                    state
+                }
+        }
+        return updateDetailsActionButtonState(updateCheckboxes, checkedCount > 0)
+    }
+
+    private fun updateDetailsActionButtonState(
+        details: List<JetpackListItemState>,
+        enableActionButton: Boolean
+    ): List<JetpackListItemState> {
+        return details.map { state ->
+            if (state.type == PRIMARY_ACTION_BUTTON) {
```

Comment:
❓ Is this equivalent to `if (state is ActionButtonState && !state.isSecondary)` ? If it is, I think it might be safer to use this approach, wdyt? ( or could we perhaps drop the "isSecondary" check completely?)

Reply:
`if (state is ActionButtonState) ` is sufficient ... in this case I can be opinionated because there is only 1 button on the details view. Updated in 14d3a34